lib/RoleIssuerResolver.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  DomainReader,
+  IIssuerDefinition,
+} from '@energyweb/credential-governance';
+import { IssuerResolver } from '@energyweb/vc-verification';
+import { providers, utils } from 'ethers';
+import { CacheServerClient } from './CacheServerClient';
+
+/**
+ * Resolves issuers definition by either by reading smart contract state via an Ethers provider or from ssi-hub (cache server)
+ */
+export class RoleIssuerResolver implements IssuerResolver {
+  private _domainReader: DomainReader;
+  private _cacheServerClient: CacheServerClient;
+
+  constructor(
+    domainReader: DomainReader,
+    provider: providers.Provider,
+    privateKey: string,
+    cacheServerUrl: string
+  ) {
+    this._domainReader = domainReader;
```

Comment:
Similar to CredentialResolver, I think we use an instance of `EthersProviderIssuerResolver ` to reduce the duplication of this code